MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -766,6 +796,12 @@ public final class MagazineLayout: UICollectionViewLayout {
   private var footerLayoutAttributes = [ElementLocation: MagazineLayoutCollectionViewLayoutAttributes]()
   private var backgroundLayoutAttributes = [ElementLocation: MagazineLayoutCollectionViewLayoutAttributes]()
 
+  // These properties are used to keep the layout attributes copies used for insert/delete
+  // animations up-to-date as items are self-sized. If we don't keep these copies up-to-date, then
+  // animations will start from the estimated height.
+  private var itemLayoutAttributesForInFlightAnimations = [IndexPath: UICollectionViewLayoutAttributes]()
+  private var supplementaryViewLayoutAttributesForInFlightAnimations = [IndexPath: UICollectionViewLayoutAttributes]()
```

Comment:
Hmm, it seems like we shouldn't hold onto these forever. It doesn't seem like we'd need the reference to them after they received the preferred heights. Or maybe we could clear them all in `finalizeCollectionViewUpdates`?

Reply:
(they are cleared, left a comment in that part of the diff)

Reply:
Ah, sorry. I missed that.